From bug report #773872
When I add hyphenation hints (hy\-phe\-na\-tion) to bibliography entries,
running biber and then latex again causes the following error:

! Package inputenc Error: Unicode char \u8:‑ not set up for use with LaTeX.

Reportedly, this bug is fixed in the 2.0 version of biber. But maybe the
corresponding patch (commit 732ebc5962850416178d4802a44583e3cd1b3740
upstream) could be backported to jessie? More information on this bug may be
found upstream at https://github.com/plk/biber/issues/46 .
---
 lib/Biber/LaTeX/Recode.pm       |    4 ++++
 lib/Biber/LaTeX/recode_data.xml |    3 +++
 t/utils.t                       |    4 ++--
 3 files changed, 9 insertions(+), 2 deletions(-)

--- biber.orig/lib/Biber/LaTeX/Recode.pm
+++ biber/lib/Biber/LaTeX/Recode.pm
@@ -129,6 +129,10 @@
         $remap_d->{$type}{map}{NFD($from->textContent())} = NFD($to->textContent());
       }
     }
+    # Things we don't want to change when decoding as this breaks some things
+    foreach my $d ($xpc->findnodes('/texmap/decode_exclude/char')) {
+      delete($remap_d->{$type}{map}{NFD($d->textContent())});
+    }
   }
 
   # Construct encode set
--- biber.orig/lib/Biber/LaTeX/recode_data.xml
+++ biber/lib/Biber/LaTeX/recode_data.xml
@@ -856,6 +856,9 @@
     <map><from>Psi</from>        <to hex="3A8">Ψ</to></map>
     <map><from>Omega</from>      <to hex="3A9">Ω</to></map>
   </maps>
+  <decode_exclude>
+    <char>-</char>
+  </decode_exclude>
   <encode_exclude>
     <char>"</char>
     <char>#</char>
--- biber.orig/t/utils.t
+++ biber/t/utils.t
@@ -4,7 +4,7 @@
 use utf8;
 no warnings 'utf8' ;
 
-use Test::More tests => 34;
+use Test::More tests => 35;
 use Biber;
 use Biber::Entry::Name;
 use Biber::Entry::Names;
@@ -88,7 +88,7 @@
 is( latex_encode(NFD('Þ')), '{\TH}', 'latex encode 6'); # Testing preferred
 is( latex_encode('$'), '$', 'latex encode 7'); # Testing exclude
 is( latex_encode(NFD('–')), '--', 'latex encode 8'); # Testing raw
-
+is( latex_decode('a\-a'), 'a\-a', 'discretionary hyphens');
 
 my @arrayA = qw/ a b c d e f c /;
 my @arrayB = qw/ c e /;
